pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1309,3 +1341,124 @@ func initializeAnnotations(cd *hivev1.ClusterDeployment) {
 		cd.Annotations = map[string]string{}
 	}
 }
+
+// mergePullSecrets merges the global pull secret JSON (if defined) with the cluster's pull secret JSON (if defined)
+// An error will be returned if neither is defined
+func (r *ReconcileClusterDeployment) mergePullSecrets(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (string, error) {
+	var localPullSecret string
+	var err error
+
+	// For code readability let's call the pull secret in cluster deployment config as local pull secret
+	if cd.Spec.PullSecret != nil {
+		localPullSecret, err = controllerutils.LoadSecretData(r.Client, cd.Spec.PullSecret.Name, cd.Namespace, corev1.DockerConfigJsonKey)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				return "", err
+			}
+		}
+	}
+
+	// Check if global pull secret from env as it comes from hive config
+	globalPullSecret := os.Getenv("GLOBAL_PULL_SECRET")
+
+	switch {
+	case globalPullSecret != "" && localPullSecret != "":
+		// Merge local pullSecret and globalPullSecret. If both pull secrets have same registry name
+		// then the merged pull secret will have registry secret from local pull secret
+		pullSecret, err := controllerutils.MergeJsons(globalPullSecret, localPullSecret, cdLog)
+		if err != nil {
+			errMsg := "unable to merge global pull secret with local pull secret"
+			cdLog.WithError(err).Error(errMsg)
+			return "", errors.Wrap(err, errMsg)
+		}
+		return pullSecret, nil
+	case globalPullSecret != "":
+		return globalPullSecret, nil
+	case localPullSecret != "":
+		return localPullSecret, nil
+	default:
+		errMsg := "clusterdeployment must specify pull secret since hiveconfig does not specify a global pull secret"
+		cdLog.Error(errMsg)
+		return "", errors.New(errMsg)
+	}
+}
+
+// updatePullSecretInfo adds pull secret information in cluster deployment and cluster deployment status
+func (r *ReconcileClusterDeployment) updatePullSecretInfo(pullSecret string, cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	var existingSecretName string
+	var err error
+
+	// Get existing pull secret object. We need the object as we might delete it later
+	pullSecretObjExists := true
+	existingPullSecretObj := &corev1.Secret{}
+	if len(cd.Status.PullSecret.Name) == 0 {
+		cdLog.Info("No existing merged pull secret in cluster deployment status")
+		pullSecretObjExists = false
```

Comment:
Both `if len(cd.Status.PullSecret.Name) == 0 ` , `if cd.Status.PullSecret.Name == "" `  returns true here even of there is `cd.Status.PullSecret.Name` exists. No idea what is going on. Because of which the installjob gets deleted to create a new install job in a hot loop.

Reply:
It still have the issue.

Reply:
@LalatenduMohanty I was not able to reproduce the issue with this code. However, it could be caused by more than one controller running at the same time. In your environment, are you by any chance running the hive controller in a pod as well as with `make run`?

Reply:
I am testing it on Minishift and here are the steps

* https://github.com/openshift/hive/blob/master/docs/install.md#deploy-using-minishift
```
$ IMG=quay.io/lalatendum/hive-devel:latest make buildah-push

$ DEPLOY_IMAGE=quay.io/lalatendum/hive-devel:latest make deploy

$ bin/hiveutil create-cluster --base-domain=new-installer.openshift.com ${CLUSTER_NAME}
```
I have tested the master with the same workflow and everything seems to work, so assuming something wrong with my code.